robotoff/app/core.py

```diff
@@ -132,6 +138,39 @@ def get_insights(
     return query.iterator()
 
 
+def get_images(
+    with_predicted: Optional[bool] = False,
+    barcode: Optional[str] = None,
+    server_domain: Optional[str] = None,
+    offset: Optional[int] = None,
+    count: bool = False,
+    limit: Optional[int] = 25,
+) -> Iterable[ImageModel]:
+    if server_domain is None:
+        server_domain = settings.OFF_SERVER_DOMAIN
+
+    where_clauses = [ImageModel.server_domain == server_domain]
+
+    if barcode:
+        where_clauses.append(ImageModel.barcode == barcode)
+
+    query = ImageModel.select()
+
+    if with_predicted:
+        # return all images
```

Comment:
It's the other way around (while the query below is ok)

```suggestion
    if not with_predicted:
        # return only images without prediction
```